pytorch_lightning/utilities/parsing.py

```diff
@@ -174,75 +174,60 @@ def __repr__(self):
-        the old hparams namespace/dict, and the datamodule. """
+def lightning_get_attr_holder(model, attribute):
+    """ Gets the object or dict that holds attribute, or None. Checks for attribute in model namespace,
+            the old hparams namespace/dict, and the datamodule, returns the first one that has it. """
     trainer = model.trainer
 
+    holder = None
+
     # Check if attribute in model
     if hasattr(model, attribute):
-        attr = True
+        holder = model
+
     # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
+    if holder is None and hasattr(model, 'hparams'):
+        if attribute in model.hparams:
+            holder = model.hparams
+
     # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
-        attr = False
+    if holder is None and trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
+        holder = trainer.datamodule
 
-    return attr
+    return holder
 
 
-def lightning_getattr(model, attribute):
-    """ Special getattr for lightning. Checks for attribute in model namespace,
+def lightning_hasattr(model, attribute):
+    """ Special hasattr for lightning. Checks for attribute in model namespace,
         the old hparams namespace/dict, and the datamodule. """
-    trainer = model.trainer
+    return lightning_get_attr_holder(model, attribute) is not None
 
-    # Check if attribute in model
-    if hasattr(model, attribute):
-        attr = getattr(model, attribute)
-    # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = model.hparams[attribute]
-        else:
-            attr = getattr(model.hparams, attribute)
 
-    # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
+def lightning_getattr(model, attribute):
+    """ Special getattr for lightning. Checks for attribute in model namespace,
+        the old hparams namespace/dict, and the datamodule. """
+    holder = lightning_get_attr_holder(model, attribute)
+    if holder is None:
         raise ValueError(f'{attribute} is neither stored in the model namespace'
                          ' nor the `hparams` namespace/dict, nor the datamodule.')
-    return attr
+
+    if isinstance(holder, dict):
+        return holder[attribute]
+    else:
```

Comment:
You don't need this else

Reply:
Wouldn't I for `Namespace`?  You can't use `[]` on them.

Reply:
You are returning for both options so only one will execute, the correct one

https://pheanex.github.io/pylint/#R1705